web_sdk/sdk_rewriter.dart

```diff
@@ -34,29 +34,29 @@ export 'dart:_engine'
 ];
 
 final List<Replacer> engineLibraryPatterns = <Replacer>[
-  AllReplacer(RegExp(r'library\s+engine;'), 'library dart._engine;'),
-  AllReplacer(RegExp(r'''
-import\s*'../ui.dart' as ui;
-'''), r'''
-import 'dart:ui' as ui;
-'''),
-  AllReplacer(RegExp(r'''
-import\s*'package:ui/ui.dart' as ui;
-'''), r'''
+  AllReplacer(RegExp(r'library\s+engine;'), '''
+@JS()
+library dart._engine;
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:convert' hide Codec;
+import 'dart:developer' as developer;
+import 'dart:html' as html;
+import 'dart:js' as js;
+import 'dart:js_util' as js_util;
+import 'dart:_js_annotations';
+import 'dart:math' as math;
+import 'dart:svg' as svg;
+import 'dart:typed_data';
```

Comment:
Btw, this change is expected fail [sdk_rewriter_test.dart](https://github.com/flutter/engine/blob/main/web_sdk/test/sdk_rewriter_test.dart). So the tests need to be updated as well.

Reply:
Done.